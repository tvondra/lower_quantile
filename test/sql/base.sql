\set ECHO none

-- disable the notices for the create script (shell types etc.)
SET client_min_messages = 'WARNING';
\i sql/lower_quantile--1.0.0.sql
\i sql/lower_quantile--1.0.0--1.0.1.sql
\i sql/lower_quantile--1.0.1--1.0.2.sql
\i sql/lower_quantile--1.0.2--1.0.3.sql
SET client_min_messages = 'NOTICE';

\set ECHO all

SELECT lower_quantile(x, 0.01::double precision) FROM generate_series(1,1000) s(x);
SELECT lower_quantile(x, 0.01::numeric) FROM generate_series(1,1000) s(x);

SELECT lower_quantile(x, ARRAY[0.01, 0.05, 0.1, 0.9, 0.95, 0.99]::double precision[]) FROM generate_series(1,1000) s(x);
SELECT lower_quantile(x, ARRAY[0.01, 0.05, 0.1, 0.9, 0.95, 0.99]::numeric[]) FROM generate_series(1,1000) s(x);
