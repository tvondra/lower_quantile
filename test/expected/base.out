\set ECHO none
SELECT lower_quantile(x, 0.01::double precision) FROM generate_series(1,1000) s(x);
 lower_quantile 
----------------
             10
(1 row)

SELECT lower_quantile(x, 0.01::numeric) FROM generate_series(1,1000) s(x);
 lower_quantile 
----------------
             10
(1 row)

SELECT lower_quantile(x, ARRAY[0.01, 0.05, 0.1, 0.9, 0.95, 0.99]::double precision[]) FROM generate_series(1,1000) s(x);
     lower_quantile      
-------------------------
 {10,50,100,900,950,990}
(1 row)

SELECT lower_quantile(x, ARRAY[0.01, 0.05, 0.1, 0.9, 0.95, 0.99]::numeric[]) FROM generate_series(1,1000) s(x);
     lower_quantile      
-------------------------
 {10,50,100,900,950,990}
(1 row)

